uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/CRM/HRLeaveAndAbsences/BAO/LeavePeriodEntitlementTest.php

```diff
@@ -688,6 +688,111 @@ private function getEntitlementCalculationMock(
+		);
+
+		// This would deduct 2 days
+		$this->createLeaveRequestBalanceChange(
+			$periodEntitlement->type_id,
+			$periodEntitlement->contact_id,
+			$this->leaveRequestStatuses['Waiting Approval'],
+			date('YmdHis'),
+			date('YmdHis', strtotime('+1 day'))
+		);
+
+		// This would deduct 1 day
+		$this->createLeaveRequestBalanceChange(
+			$periodEntitlement->type_id,
+			$periodEntitlement->contact_id,
+			$this->leaveRequestStatuses['More Information Requested'],
+			date('YmdHis')
+		);
+
+		$this->assertEquals(4, $periodEntitlement->getFutureBalance());
+	}
+
+	public function testGetPeriodEntitlementsForContact() {
+    LeavePeriodEntitlement::create([
+      'period_id' => 1,
+      'type_id' => 1,
+      'contact_id' => 1
+    ]);
+
+    LeavePeriodEntitlement::create([
+      'period_id' => 1,
+      'type_id' => 2,
+      'contact_id' => 1
+    ]);
+
+    $entitlements = LeavePeriodEntitlement::getPeriodEntitlementsForContact($contact_id = 1, $period_id = 1);
+    $this->assertEquals(2, count($entitlements));
+
+  }
+
+  public function testgetLeavePeriodEntitlementRemainder() {
+    $absence_period = $this->createAbsencePeriod(date('Y') .'-01-01', date('Y') .'-12-31');
+    $periodEntitlement = $this->createPeriodEntitlement($absence_period->id);
+    $this->createLeaveBalanceChange($periodEntitlement->id, 10);
+
+    // This leave request will deduct 3 days from the entitlement
+    $this->createLeaveRequestBalanceChange(
+      $periodEntitlement->type_id,
+      $periodEntitlement->contact_id,
+      $this->leaveRequestStatuses['Approved'],
+      date('YmdHis'),
+      date('YmdHis', strtotime('+2 day'))
+    );
+
+    $params = ['entitlement_id' => $periodEntitlement->id];
+    $result = $periodEntitlement::getLeavePeriodEntitlementRemainder($params);
+    $this->assertEquals(1, count($result));
+    $this->assertEquals(7, $result[0]['remainder']['current']);
+  }
+
+  public function testgetLeavePeriodEntitlementRemainderWithIncludeFuture() {
+    $absence_period = $this->createAbsencePeriod(date('Y') .'-01-01', date('Y') .'-12-31');
+    $periodEntitlement = $this->createPeriodEntitlement($absence_period->id);
+    $this->createLeaveBalanceChange($periodEntitlement->id, 10);
+
+    // This leave request will deduct 3 days from the entitlement
+    $this->createLeaveRequestBalanceChange(
+      $periodEntitlement->type_id,
+      $periodEntitlement->contact_id,
+      $this->leaveRequestStatuses['Waiting Approval'],
+      date('YmdHis'),
+      date('YmdHis', strtotime('+2 day'))
+    );
+
+    $params = ['entitlement_id' => $periodEntitlement->id, 'include_future' => true];
+    $result = $periodEntitlement::getLeavePeriodEntitlementRemainder($params);
+    $this->assertEquals(1, count($result));
+    $this->assertEquals(10, $result[0]['remainder']['current']);
+    $this->assertEquals(7, $result[0]['remainder']['future']);
+  }
+
+  public function testgetLeavePeriodEntitlementRemainderWithContactAndPeriodId() {
```

Comment:
All of you tests for the getLeavePeriodEntitlementRemainder() always deal only with the scenario where a single row is returned. It would be nice to have at least one test with multiple rows are returned, to assert that the method works correctly on that situation.